src/main/java/com/coveo/configuration/parameterstore/ParameterStorePropertySourceEnvironmentPostProcessor.java

```diff
@@ -29,15 +35,29 @@
     public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application)
     {
         if (!initialized && isParameterStorePropertySourceEnabled(environment)) {
-            environment.getPropertySources()
-                       .addFirst(new ParameterStorePropertySource(PARAMETER_STORE_PROPERTY_SOURCE_NAME,
-                                                                  new ParameterStoreSource(buildAWSSimpleSystemsManagementClient(environment),
-                                                                                           environment.getProperty(PARAMETER_STORE_HALT_BOOT_CONFIGURATION_PROPERTY,
-                                                                                                                   Boolean.class,
-                                                                                                                   Boolean.FALSE))));
-            if (!environment.getProperty(PARAMETER_STORE_SUPPORT_MULTIPLE_APPLICATION_CONTEXTS_CONFIGURATION_PROPERTY,
-                                         Boolean.class,
-                                         Boolean.FALSE)) {
+            boolean haltBoot = environment.getProperty(PARAMETER_STORE_HALT_BOOT_CONFIGURATION_PROPERTY,
+                                                       Boolean.class,
+                                                       Boolean.FALSE);
+            if (isMultiRegionEnabled(environment)) {
+                List<String> signingRegions = getSigningRegions(environment);
+                ListIterator<String> regionsIterator = signingRegions.listIterator(signingRegions.size());
+
+                while (regionsIterator.hasPrevious()) {
+                    boolean isLast = regionsIterator.hasNext();
+                    String region = regionsIterator.previous();
+
+                    environment.getPropertySources()
+                               .addFirst(buildParameterStorePropertySource(PARAMETER_STORE_PROPERTY_SOURCE_NAME
```

Comment:
add an internal method that gets a `isLast` parameter to avoir a weird ternary if(true) return false ;)